backend/src/main/java/gov/cdc/usds/simplereport/service/SlackMessageService.java

```diff
@@ -0,0 +1,121 @@
+package gov.cdc.usds.simplereport.service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.json.JSONException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.http.MediaType;
+import org.springframework.http.RequestEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestClientException;
+
+import gov.cdc.usds.simplereport.config.simplereport.DataHubConfig;
+
+/**
+ * Utility service for sending slack messages by way of the webhook configured
+ * in {@link DataHubConfig#getSlackNotifyWebhookUrl()}
+ */
+@Service
+public class SlackMessageService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SlackMessageService.class);
+
+    private DataHubConfig _config;
+    private RestTemplateBuilder _builder;
+
+    public SlackMessageService(DataHubConfig config, RestTemplateBuilder builder) {
+        _config = config;
+        _builder = builder;
+    }
+
+    public void sendSlackChannelMessage(String titleMsg, List<String> markupMsgs, Boolean separateMsgs) {
+        if (!_config.getSlackNotifyWebhookUrl().startsWith("https://hooks.slack.com/")) {
```

Comment:
The service will only be able to post into a single slack channel. We will _eventually_ need some way to have different channels webhookurls?

Reply:
Yeah, I didn't want to get ahead of myself, but I can see eventually having the method be `sendSlackChannelMessage(AlertChannelEnumeration whichChannel, String titleMsg, List<String> body)`, and having a lookup table of valid hooks—either that or switching to using the chat API, which I think lets us pick a channel at runtime? I tried not to get too far ahead of my skis on this one, though. 😄